test/CMakeLists.txt

```diff
@@ -2,290 +2,74 @@
-add_executable(${TARGET_THREAD} ${TARGET_THREAD_SOURCES})
-target_include_directories(${TARGET_THREAD} PRIVATE ${UVW_SRC_DIR})
-target_link_libraries(${TARGET_THREAD} PRIVATE ${COMMON_LINK_LIBS})
-add_test(NAME ${TARGET_THREAD} COMMAND ${TARGET_THREAD})
-
-# Test TARGET_TIMER
-
-set(TARGET_TIMER_SOURCES $<TARGET_OBJECTS:odr> uvw/timer.cpp)
-add_executable(${TARGET_TIMER} ${TARGET_TIMER_SOURCES})
-target_include_directories(${TARGET_TIMER} PRIVATE ${UVW_SRC_DIR})
-target_link_libraries(${TARGET_TIMER} PRIVATE ${COMMON_LINK_LIBS})
-add_test(NAME ${TARGET_TIMER} COMMAND ${TARGET_TIMER})
-
-# Test TARGET_TTY
-
-set(TARGET_TTY_SOURCES $<TARGET_OBJECTS:odr> uvw/tty.cpp)
-add_executable(${TARGET_TTY} ${TARGET_TTY_SOURCES})
-target_include_directories(${TARGET_TTY} PRIVATE ${UVW_SRC_DIR})
-target_link_libraries(${TARGET_TTY} PRIVATE ${COMMON_LINK_LIBS})
-add_test(NAME ${TARGET_TTY} COMMAND ${TARGET_TTY})
-
-# Test TARGET_UDP
-
-set(TARGET_UDP_SOURCES $<TARGET_OBJECTS:odr> uvw/udp.cpp)
-add_executable(${TARGET_UDP} ${TARGET_UDP_SOURCES})
-target_include_directories(${TARGET_UDP} PRIVATE ${UVW_SRC_DIR})
-target_link_libraries(${TARGET_UDP} PRIVATE ${COMMON_LINK_LIBS})
-add_test(NAME ${TARGET_UDP} COMMAND ${TARGET_UDP})
-
-# Test TARGET_UTIL
-
-set(TARGET_UTIL_SOURCES $<TARGET_OBJECTS:odr> uvw/util.cpp)
-add_executable(${TARGET_UTIL} ${TARGET_UTIL_SOURCES})
-target_include_directories(${TARGET_UTIL} PRIVATE ${UVW_SRC_DIR})
-target_link_libraries(${TARGET_UTIL} PRIVATE ${COMMON_LINK_LIBS})
-add_test(NAME ${TARGET_UTIL} COMMAND ${TARGET_UTIL})
-
-# Test TARGET_WORK
+# List of available targets
 
-set(TARGET_WORK_SOURCES $<TARGET_OBJECTS:odr> uvw/work.cpp)
-add_executable(${TARGET_WORK} ${TARGET_WORK_SOURCES})
-target_include_directories(${TARGET_WORK} PRIVATE ${UVW_SRC_DIR})
-target_link_libraries(${TARGET_WORK} PRIVATE ${COMMON_LINK_LIBS})
-add_test(NAME ${TARGET_WORK} COMMAND ${TARGET_WORK})
+ADD_UVW_TEST(main main.cpp)
+ADD_UVW_TEST(async uvw/async.cpp)
+ADD_UVW_TEST(check uvw/check.cpp)
+ADD_UVW_TEST(dns uvw/dns.cpp)
+ADD_UVW_TEST(emitter uvw/emitter.cpp)
+ADD_UVW_DIR_TEST(file_req uvw/file_req.cpp)
+ADD_UVW_DIR_TEST(fs_event uvw/fs_event.cpp)
+ADD_UVW_DIR_TEST(fs_poll uvw/fs_poll.cpp)
+ADD_UVW_DIR_TEST(fs_req uvw/fs_req.cpp)
+ADD_UVW_TEST(handle uvw/handle.cpp)
+ADD_UVW_TEST(idle uvw/idle.cpp)
+ADD_UVW_TEST(lib uvw/lib.cpp)
+ADD_UVW_TEST(loop uvw/loop.cpp)
+ADD_UVW_TEST(pipe uvw/pipe.cpp)
+ADD_UVW_TEST(poll uvw/poll.cpp)
+ADD_UVW_TEST(prepare uvw/prepare.cpp)
+ADD_UVW_TEST(process uvw/process.cpp)
+ADD_UVW_TEST(request uvw/request.cpp)
+ADD_UVW_TEST(resource uvw/resource.cpp)
+ADD_UVW_TEST(signal uvw/signal.cpp)
+ADD_UVW_TEST(stream uvw/stream.cpp)
+ADD_UVW_TEST(tcp uvw/tcp.cpp)
+ADD_UVW_TEST(thread uvw/thread.cpp)
+ADD_UVW_TEST(timer uvw/timer.cpp)
+ADD_UVW_TEST(tty uvw/tty.cpp)
+ADD_UVW_TEST(udp uvw/udp.cpp)
+ADD_UVW_TEST(util uvw/util.cpp)
+ADD_UVW_TEST(work uvw/work.cpp)
+
+# Test lib
+
+target_link_libraries(lib PRIVATE ${CMAKE_DL_LIBS})
```

Comment:
I'd rather add a third macro or a dedicated function. Otherwise stuff is spread around the file and it's hard to understand with a glimpse what's going on under the hood.